javascript/ql/src/semmle/javascript/security/dataflow/IncompleteHtmlAttributeSanitizationCustomizations.qll

```diff
@@ -51,8 +51,11 @@ module IncompleteHtmlAttributeSanitization {
     string lhs;
 
     HtmlAttributeConcatenation() {
-      lhs = this.getPreviousLeaf().getStringValue().regexpCapture("(.*)=\"[^\"]*", 1) and
-      this.getNextLeaf().getStringValue().regexpMatch(".*\".*")
+      lhs = this.getPreviousLeaf().getStringValue().regexpCapture("((?:[\n\r]|.)*)=\"[^\"]*", 1) and
+      (
+        this.getNextLeaf().getStringValue().regexpMatch(".*\".*") or
+        this.getRoot().getConstantStringParts().regexpMatch("(?:[\n\r]|.)*</.*")
```

Comment:
The right way to set the dotall flag is to put `(?s)` in the beginning of the regexp (see `HtmlConcatenationRoot`). But yes, it's a rather surprising default.

Speaking of `HtmlConcatenationRoot` could that be used in the second condition instead?

Reply:
> The right way to set the dotall flag is to put (?s) in the beginning of the regexp (see HtmlConcatenationRoot). But yes, it's a rather surprising default.

:+1: 

> Speaking of HtmlConcatenationRoot could that be used in the second condition instead?

I'm not sure what you are suggesting. Something like `this.getNextLeaf+().getStringValue().regexpMatch("(?s).*</.*")`?  
(That would require removing `pragma[inline]` from `getNextLeaf()`).